markdown/inlinepatterns.py

```diff
@@ -151,7 +151,7 @@ def build_inlinepatterns(md_instance, **kwargs):
 AUTOMAIL_RE = r'<([^> \!]*@[^> ]*)>'
 
 # <...>
-HTML_RE = r'(\<([a-zA-Z/][^\>]*?|\!--.*?--)\>)'
+HTML_RE = r'(\<([a-zA-Z/][^\>]*?|\!--.*?--)\>)(?!\>)'
```

Comment:
I think this is a sensible approach when the smarty extension is in use, but the default behavior should not be changed when smarty is not enabled. I would expect that the smarty extension would replace the HTML pattern with a new one if angle quotes are enabled. Otherwise, leave it unchanged.

And we should have a test for each state (enabled and not enabled). The not enabled state should return the same results as we would get without smarty before this patch was applied.